### PR TITLE
Zellic audit #2 feedback

### PR DESCRIPTION
**Pull request by neodaoist**
- Semantic
  - Audit 2 feedback 3.1 — Remove `settlementSeed` updating in `_assignExercise()` function
  - Audit 2 feedback 3.2 — We have an elegant solution in development (https://github.com/valorem-labs-inc/LibDDRV), but it will not be included as part of our v1 release
  - Audit 2 Telegram feedback — Support off-chain claim accounting via bucket-related events (necessitated modification of `_assignExercise()` function signature, to pass in `optionId` parameter)
  - Correct the mutability of `feeBps` from immutable to constant, suppressing `const-name-snakecase` solhint warning (https://github.com/valorem-labs-inc/valorem-core/pull/201/files#diff-033680dc4cc50d115cdec7fe97c8be2fcf62aacef3c4b50b7c9979c08ad0027fR102)
- Syntactic
  - Rename `OptionSettlementEngine` to `ValoremOptionsClearinghouse`
  - Bump copyright year
  - Clarify and fix typos on few comments
  - Add ASCII art and top-line NatSpec to `IValoremOptionsClearinghouse` interface
  - Update trust model document to reflect the up-to-date implementation
- Deployment scripts
  - Support deploying to deterministic addresses
- CI
  - Re-enable integration tests
  - Exclude non-Clearinghouse tests from coverage report
  
  
_Note: you can press `a` to toggle check annotations on/off_

### CODE REVIEW

File: src/ValoremOptionsClearinghouse.sol [hunk 1]
```diff
@@ -800,20 +808,19 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
...
                 unexercisedBucketIndices.pop();
             } else {
-                // Bucket is partially exercised/assigned
+                // Bucket is partially exercised/assigned.
                 amountPresentlyExercised = amount;
                 amount = 0;
             }
             bucketInfo.amountExercised += amountPresentlyExercised;
 
+            emit BucketAssignedExercise(optionId, bucketIndex, amountPresentlyExercised);
```
Comment (Flip-Liquid) [inline src/ValoremOptionsClearinghouse.sol:817]: looks like this is the line that addresses offline accounting

**Comment by 0xAlcibiades** [comment]
> @neodaoist please review the comment I left in the impl file; feel free to resolve it if we've already had that discussion.
> 
> other than that, it seems like the description of the PR is slightly off; the seed update is not included in the delta for this PR

They should be.

File: src/ValoremOptionsClearinghouse.sol (hunk 1 above)
Comment (neodaoist) [reply src/ValoremOptionsClearinghouse.sol:817]: Right, but without the `BucketWrittenInto` event, an off-chain actor won't be able to make heads or tails of what this event means. They need to know how many options, for each claim/index, are being written into the buckets to support off-chain claim accounting.

**Comment by neodaoist** [comment]
> @neodaoist please review the comment I left in the impl file; feel free to resolve it if we've already had that discussion.
> 
> other than that, it seems like the description of the PR is slightly off; the seed update is not included in the delta for this PR

https://github.com/valorem-labs-inc/valorem-core/pull/199/files#diff-033680dc4cc50d115cdec7fe97c8be2fcf62aacef3c4b50b7c9979c08ad0027fL813

**Review by 0xAlcibiades — APPROVED** [review]

**Pull request merged**